### PR TITLE
Fix cmake warnings

### PR DESCRIPTION
**Pull request by henningkayser**
This fixes the build warnings discovered here https://github.com/ros-planning/moveit2/runs/3662305871?check_suite_focus=true#step:8:689. Should we enforce the build to be free of warnings?

### CODE REVIEW

**Review by destogl — APPROVED** [review]
This looks good. I suggest renaming the PR to something like "Fix build warnings". I found "cmake" in the title somewhat misleading.

There is also one more `clang-tidy` issue.

**Review by JafarAbdi — COMMENTED** [review]
> Should we enforce the build to be free of warnings?

+1, the only problem with this is it may force us to branch off galactic since some features are deprecated in rolling and the alternatives don't exist in galactic

**Comment by v4hn** [comment]
> Should we enforce the build to be free of warnings

We do that [for moveit](https://github.com/ros-planning/moveit/blob/master/.github/workflows/ci.yaml#L41), so I think it's a very good idea to enforce it for moveit2 as well!

> +1, the only problem with this is it may force us to branch off galactic since some features are deprecated in rolling and the alternatives don't exist in galactic

I would argue that using deprecated API should still be avoided in the newer distro, possibly using `#ifdef` approaches to support both alternatives.
That way, you can add clear TODOs to address when galactic support is dropped in the branch.

File: moveit_kinematics/test/test_kinematics_plugin.cpp [hunk 1]
```diff
@@ -477,8 +476,8 @@ TEST_F(KinematicsTest, unitIK)
   };
 
   std::vector<double> ground_truth, pose_values;
-  constexpr char POSE_TYPE_RELATIVE[] = "relative";
-  constexpr char POSE_TYPE_ABSOLUTE[] = "absolute";
+  constexpr char post_type_relative[] = "relative";
```
Comment (DLu) [inline moveit_kinematics/test/test_kinematics_plugin.cpp:479]: Note: My PR was dumb and had a typo. Should be pose_type_relative (not post)

**Review by JafarAbdi — APPROVED** [review]

**Pull request merged**